dash-pipeline/SAI/templates/saiapi.h.j2

```diff
@@ -34,7 +34,7 @@
  */
 
 {% for table in sai_api.tables %}
-{% if table.actions | length > 1 or ((table.actions | length == 1) and (((table.is_object == 'false') or (table['keys'] | length <= 1)) and ((table['action_params'] | length == 0))))  %}
+{% if table.actions | length > 1 or table.is_object == 'false' %}
```

Comment:
Original check tries to generate action-type even for sai object if it has only 1 action with 0 params. Also it skips 0 action case. If these are no longer needed / not possible, then this simpler should be fine.

Reply:
yea, there are 2 cases here:

- Case 1: Only 1 action with 0 parameters - this is added to cover the VIP and PA validation case for permit action, which doesn't have any parameters. Both VIP and PA validation are entries not objects, so `is_object == false` will get them covered there.
- Case 2: 0 action case. This can only happen if a P4 table has only 1 defaultonly action. In this case, it is better to directly apply the action instead of introducing a table.

so to me, it looks these two cases are both ok to be simplified. what do you think?

Reply:
sure

Reply:
also for the case 1, 0 parameter is ok, as we can see after the check is removed, the SAI API generation works and generated API looks the same.

Reply:
thanks Mukesh. as you approved this PR, I am resolving the comment now.